File: Python/1389.py
# 1389. Create Target Array in the Given Order
# https://leetcode.com/problems/create-target-array-in-the-given-order/

class Solution:
  def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
    out = []

    for (i, v) in enumerate(nums):
      out.insert(index[i], v)

    return out

File: Rust/1389.rs
/**
 * 1389. Create Target Array in the Given Order
 * https://leetcode.com/problems/create-target-array-in-the-given-order/
 */

impl Solution {
  pub fn create_target_array(nums: Vec<i32>, index: Vec<i32>) -> Vec<i32> {
    let mut out: Vec<i32> = Vec::new();

    for (i, v) in nums.into_iter().enumerate() {
      out.insert(index[i] as usize, v);
    }

    return out;
  }
}
